src/services/recentlyViewedProjects.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+angular.module("openshiftCommonServices")
+  .service("RecentlyViewedProjectsService", function(){
+
+    var recentlyViewedProjsKey = "common-recently-viewed-projects";
+
+    var getProjects = function() {
+      var recentlyViewed = localStorage.getItem(recentlyViewedProjsKey);
+      return recentlyViewed ? JSON.parse(recentlyViewed) : [];
+    };
+
+    var addProject = function(uid) {
+      var recentlyViewed = getProjects();
+
+      // if previously viewed, remove from list
+      _.remove(recentlyViewed, function(rvUID) {
+        return rvUID === uid;
+      });
+
+      // add to front of list
+      recentlyViewed.unshift(uid);
+
+      // limit to 5 items
+      recentlyViewed = _.take(recentlyViewed, 5);
+
+      setRecentlyViewedProjects(recentlyViewed);
+    };
+
+    var clear = function() {
+      localStorage.removeItem(recentlyViewedProjsKey);
+    };
+
+    var setRecentlyViewedProjects = function(recentlyViewed) {
+      localStorage.setItem(recentlyViewedProjsKey, JSON.stringify(recentlyViewed));
+    };
+
+    var orderByMostRecentlyViewed = function(projects) {
+      var recentlyViewedProjects = [];
+      var recentlyViewedIds = getProjects();
+
+      _.forEach(recentlyViewedIds, function(uid) {
+        var proj = _.remove(projects, function(project) {
+          return project.metadata.uid === uid;
+        })[0];
+        if(proj !== undefined) {
+          recentlyViewedProjects.push(proj);
+        }
+      });
+
+      return recentlyViewedProjects.concat(projects);
```

Comment:
Though we agreed that it would be ordered by 'most recently viewed' and then by 'most recently created'.  Are we changing our minds on this?